pkg/runtime/processor/components.go

```diff
@@ -188,6 +191,8 @@ func (p *Processor) processComponents(ctx context.Context) error {
 
 // WaitForEmptyComponentQueue waits for the component queue to be empty.
 func (p *Processor) WaitForEmptyComponentQueue() {
+	p.pendingComponentsWaitingLock.Lock()
+	defer p.pendingComponentsWaitingLock.Unlock()
 	p.pendingComponentsWaiting.Wait()
```

Comment:
Why don't we change `pendingComponentsWaiting` into a RWMutx, where `AddPendingComponent` does an `RLock` and `WaitForEmptyComponentQueue` does a `.Lock()`.

Reply:
ig, what's the difference tho?

Reply:
We currently have 2 variables trying to do the same thing, it seems confused.